articles/using-picasso-in-android/index.md

```diff
@@ -0,0 +1,382 @@
+ switch (view.getId()) {
+	 case R.id.showDrawable:
+	 Picasso.get().load(R.drawable.image).into(imageView);
+	 break;
+```
+### d)Placeholder
+A placeholder usually is a drawable image displayed before an image is loaded into an image view. This feature comes in handy, especially if an image takes time to be loaded. In our `MainActivity.java`, in the `initializeView` method, add the following lines of code;
+
+```java
+buttonPlaceholderImage = (Button) findViewById(R.id.showPlaceholder); buttonPlaceholderImage.setOnClickListener(this);
+
+ case R.id.showPlaceholder:
+	 Picasso.get().load(R.drawable.placeholder).into(imageView);
+	 break;
+```
+### e)Callback methods.
+Picasso provides callback methods to keep track and show a loaded image's status and display a text/toast accordingly. We have displayed a toast message to show either an image is loaded successfully or an error. In our `MainActivity.java`, in the `initializeView` method, add the following lines of code;
+```java
+ buttonCallback = (Button) findViewById(R.id.showCallBack); buttonCallback.setOnClickListener(this); 
+
+            case R.id.showCallback:
+                Picasso.get().load("www.google.com").error(R.mipmap.ic_launcher).into(imageView, new Callback() {
+                    @Override
+                    public void onSuccess() {
+                        Log.d("TAG", "onSuccess");
+                    }
+                    @Override
+                    public void onError(Exception exception) {
+                        Toast.makeText(getApplicationContext(), "Cannot fetch data error", Toast.LENGTH_SHORT).show();
+                    }
+                });
+                break;
+```
+### f)Errors
+An error drawable is usually displayed when an image is not loaded successfully. We use the `error()` method. In our `MainActivity.java`, in the `initializeView` method, add the following lines of code; 
+
+```java
+buttonErrorImage = (Button) findViewById(R.id.showError); buttonErrorImage.setOnClickListener(this); 
+
+ case R.id.showError:
+	 Picasso.get().load("www.google.com").placeholder(R.drawable.placeholder).error(R.drawable.error).into(imageView);
+	 break;
+```
+### g)Rotate
+This refers to changing the orientation of an image by degrees from a point (0,0). A `rotate()` method is normally used. 
+```java
+buttonRotateImage = (Button) findViewById(R.id.showRotate); buttonRotateImage.setOnClickListener(this); 
+
+case R.id.showRotate: 
+	 Picasso.get().load(R.drawable.image).rotate(90f).into(imageView);
+	 break;
+```  
+### h)Targets  
+Targets combine image loading, callbacks, and errors. They return bitmap images. Targets normally use `onBitmapLoaded()`, `onBitmapFailed()` and on `Prepared()` methods. In our `MainActivity.java`, inside the `initializeView` method, add the following lines of code:
+```java
+buttonTarget = (Button) 
+findViewById(R.id.showTarget); buttonTarget.setOnClickListener(this); 
+
+ case R.id.showTarget:
```

Comment:
I think this case block goes to the `onClick` method. Make sure you separate these so as not to confuse the reader.
The same applies to other code blocks